friends/src/components/FriendForm.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react'
+
+export default class FriendForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state={
+            friend: this.props.activeFriend || {
+                name: '',
+                age: '',
+                email: ''
+            }
+        }
+
+    }
+
+    componentDidUpdate(prevProps) {
+        if(
+            this.props.activeFriend &&
+            prevProps.activeFriend !== this.props.activeFriend
+        ) {
+            this.setState({
+                friend: this.props.activeFriend
+            })
+        }
+     }
+
+    changeHandler = e => {
+        e.persist();
+        this.setState(prevState => ({
+            friend: {
+                ...prevState.friend,
+                [e.target.name]: e.target.value
+                }
+        }))
+    };
+
+    handleSubmit = e => {
+        e.preventDefault();
+        this.props.addFriend(this.state.friend)
+    }
+
+
+  render() {
+    return (
+        <div className="form-container">
+            <form onSubmit={this.handleSubmit}>
```

Comment:
Then onSubmit, we can decide which method to call when there is an activeFriend in our state
```suggestion
            <form onSubmit={this.props.activeFriend ? this.handleUpdate : this.handleSubmit}>
```